pkg/utils/helper.go

```diff
@@ -0,0 +1,38 @@
+// Copyright © 2024 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "github.com/bank-vaults/secret-init/pkg/provider"
+
+// Function to check if a key is duplicated in the secrets slice
+func IsKeyDuplicated(secrets *[]provider.Secret, searchKey string) bool {
+	for _, secret := range *secrets {
```

Comment:
lets add nil checks here just in case

```suggestion
	if secrets == nil {
		return false
	}
	
	for _, secret := range *secrets {
```